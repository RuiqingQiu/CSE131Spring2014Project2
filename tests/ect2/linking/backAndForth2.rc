const int c = 10;
extern function : void back(int x);

function : void forth(int x){
   static bool t = true;
   static int i = 0;
   if (i == c) {
      return;
   }

   if (t){
      t = !t;
      back(x+1);
   }
   else{
      t = !t;
      cout << " and forth " << x <<  endl;
      i++;
      forth(x);
   }
}

function : int main(){
   forth(1);
   return 0;
}
