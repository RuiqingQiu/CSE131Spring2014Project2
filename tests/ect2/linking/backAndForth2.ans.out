back 2 and forth 3
back 4 and forth 5
back 6 and forth 7
back 8 and forth 9
back 10 and forth 11
back 12 and forth 13
back 14 and forth 15
back 16 and forth 17
back 18 and forth 19
back 20 and forth 21
